cache/src/accessible_ext.rs

```diff
@@ -0,0 +1,300 @@
+		&self,
+	) -> impl Future<Output = Result<Vec<i32>, Self::Error>> + Send;
+	fn get_siblings_before<'a>(
+		&self,
+	) -> impl Future<Output = Result<Vec<AccessibleProxy<'a>>, Self::Error>> + Send
+	where
+		Self: Sized;
+	fn get_siblings_after<'a>(
+		&self,
+	) -> impl Future<Output = Result<Vec<AccessibleProxy<'a>>, Self::Error>> + Send
+	where
+		Self: Sized;
+	fn get_children_caret<'a>(
+		&self,
+		after: bool,
+	) -> impl Future<Output = Result<Vec<AccessibleProxy<'a>>, Self::Error>> + Send
+	where
+		Self: Sized;
+	/* TODO: not sure where these should go since it requires both Text as a self interface and
+	 * Hyperlink as children interfaces. */
+	fn get_next<'a>(
+		&self,
+		role: Role,
+		backward: bool,
+		already_visited: &'a mut Vec<ObjectRef>,
+	) -> impl Future<Output = Result<Option<AccessibleProxy<'a>>, Self::Error>> + Send
+	where
+		Self: Sized;
+	/// Get all edges for a given accessible object.
+	/// This means: all children, siblings, and parent, in that order.
+	/// If a direction is specified, then it will only get the appicable matching siblings/children.
+	/// This also checks if the element supports the text interface, and then checks if the caret position is contained within the string, if it is, then children are also handled by direction.
+	fn edges<'a>(
+		&self,
+		backward: Option<bool>,
+	) -> impl Future<Output = Result<Vec<AccessibleProxy<'a>>, Self::Error>> + Send
+	where
+		Self: Sized;
+	fn get_relation_set_ext<'a>(
+		&self,
+	) -> impl Future<
+		Output = Result<HashMap<RelationType, Vec<AccessibleProxy<'a>>>, Self::Error>,
+	> + Send
+	where
+		Self: Sized;
+	fn match_(&self, role: Role) -> impl Future<Output = Result<bool, OdiliaError>> + Send;
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub trait AccessibleExtError: Convertable {
+	type Error: std::error::Error
+		+ From<<Self as Convertable>::Error>
+		// TODO: add all convertable error types
+		+ From<std::num::TryFromIntError>
+		+ Send;
+}
+impl AccessibleExtError for AccessibleProxy<'_> {
+	type Error = Error;
+}
+
+impl AccessibleExt for AccessibleProxy<'_> {
+	type Error = OdiliaError;
+	async fn get_application_ext<'a>(&self) -> Result<AccessibleProxy<'a>, Self::Error>
+	where
+		Self: Sized,
+	{
+		let or: ObjectRef = self.get_application().await?;
+		let io: AccessiblePrimitive = or.into();
+		Ok(io.into_accessible(self.connection()).await?)
+	}
+	async fn get_parent_ext<'a>(&self) -> Result<AccessibleProxy<'a>, Self::Error>
+	where
+		Self: Sized,
+	{
+		let or: ObjectRef = self.parent().await?;
+		let io: AccessiblePrimitive = or.into();
+		Ok(io.into_accessible(self.connection()).await?)
+	}
+	async fn get_children_indexes<'a>(&self) -> Result<Vec<i32>, Self::Error> {
+		let mut indexes = Vec::new();
+		for child in self.get_children_ext().await? {
+			indexes.push(child.get_index_in_parent().await?);
+		}
+		Ok(indexes)
+	}
+	async fn get_children_ext<'a>(&self) -> Result<Vec<AccessibleProxy<'a>>, Self::Error>
```

Comment:
The _ext methods specifically return usable AccessibleProxy's. This will be eased by https://github.com/odilia-app/atspi/pull/182, which will be incorporated in a future PR